core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -562,6 +564,91 @@ public static List<ByteString> split(ByteString s, ByteString delimiter) {
     }
   }
 
+  public static @Nullable Boolean containsSubstr(@Nullable Object @Nullable [] s0, String s1) {
```

Comment:
Each of these methods needs a simple javadoc comment. Without such comments, it's quite difficult to map from java types to the SQL types.